src/patches/dense_patch_computation.cpp

```diff
@@ -120,26 +119,22 @@ void stitch_boundaries(
 }
 
 
-void apply_routing_region(DenseSlice& slice, const RoutingRegion& routing_region)
-{
-    for (const auto& occupied_cell : routing_region.cells)
-        slice.place_sparse_patch(SparsePatch{{PatchType::Routing, PatchActivity::None},occupied_cell}, false);
-}
-
-void mark_busy_region(DenseSlice& slice, const RoutingRegion& routing_region)
+void apply_routing_region(
+    DenseSlice& slice, 
+    const RoutingRegion& routing_region,
+    bool fail_on_already_busy_cell = true)
 {
     for (const auto& occupied_cell : routing_region.cells)
     {
         auto& patch = slice.patch_at(occupied_cell.cell);
-        if (patch)
+        if (!fail_on_already_busy_cell && patch)
```

Comment:
Recreates `mark_busy_region` behavoiur of ignoring the previously occupied region